### ISSUE TITLE
Citation database updates and concerns

### ISSUE THREAD

**Issue by gt150** [ResetNetwork/recoding-tech#76]
Before we can launch we need to figure out how to update existing citations. I've brought this up in another issue, but there at times that I need to edit/delete a citation, add a tag, or edit the existing tags. Right now the update from Zotero does not account for this, requiring me to make the changes both in Zotero and the citation database on the CMS. I believe Josh was trying to come up with a fix, but if it's not possible then maybe it makes sense to just stop using Zotero for now, and just use the citation database in the CMS. I don't really like needing to manually add citation, but the Zotero citation importer tool doesn't always work and given that were only capturing a handful of fields, I could live with it for now.  We just need to figure out the solution.

**Comment by danblah** [comment]
Related to this https://github.com/ResetNetwork/recoding-tech/issues/43

**Comment by jheretic** [comment]
Just as an update (sorry for the delay, I've been functionally out for much of the last couple weeks due to family and internet issues) I've completed the work on the import script that I believe should fix this. I'm just testing it thoroughly before deploying it to make sure it doesn't disrupt existing data.

**Comment by gt150** [comment]
@jheretic thanks for the update and work. Sorry to hear about your family issues. If for some reason the fix doesn't work or proves to complex, as I indicated above I can live with relying on the CMS citation database to get us to launch. Getting #23 addressed is also critical for launch and I wonder if it might be easier solved if we only rely on the CMS. Just offering some suggestions given your availability constraints and Reset's desire to see this launched sooner than later.

**Comment by jheretic** [comment]
Thanks. I've actually concluded development on this already and I'm preparing a big pull request that will address this as well as #23, along with some other cleanup and optimization. That should be up for testing in the next day or so. So good suggestion but no need to change tactics at this point.

**Comment by gt150** [comment]
@jheretic @rudietuesdays I'm finding s issues with the update. When I go to the library page and use the buttons it now brings up an incomplete list of results. For example, the results for disinformation start with citations from July 2021, when there are much more recent results in the citation database. 
<img width="1286" alt="Screen Shot 2022-02-22 at 2 01 44 PM" src="https://user-images.githubusercontent.com/9727494/155200879-f64187ae-4328-4d5d-bd4e-dded13d40128.png">

Also the list includes items that I deleted from Zotero. The citation database is still listing it, even though its not in Zotero and it does not have any topics associated with it. 
<img width="1286" alt="Screen Shot 2022-02-22 at 2 01 44 PM" src="https://user-images.githubusercontent.com/9727494/155201175-11d18474-675c-4a1d-9631-969708283147.png">

**Comment by harumhelmy** [comment]
Hi @gt150! We pushed a fix to the filtering issue, but currently having issues with the test site, unfortunately. @jheretic is working on a fix for that now, we'll let you know if we manage to fix it in the next hour, otherwise we'll get back to you Monday morning.

**Comment by harumhelmy** [comment]
@gt150 Sorry for the delay -- I had a bit of a disagreement with the test site all day, but we've worked through it 😅. The filtering should work properly now on the search page! Try it here: https://test.recoding.tech/search. Additionally, if you go to any topic page, the View All link on the Latest Headlines and Highlights component should pre-filter the search page. Try it out: https://test.recoding.tech/country/united-states. If this looks good to you, we can merge it with the production site

@jheretic is working on updating the imports script once more to remove the Sanity items that you've since deleted on Zotero.

**Comment by gt150** [comment]
@harumhelmy yep, looks like it working😃!!!

**Comment by jheretic** [comment]
The production site now has further import script updates enabled, so that citations and creators that aren't in Zotero will get pruned from the site automatically. This does mean that any Sanity-side updates to citations that subsequently get deleted in Zotero will be removed, and fields in Zotero will override the matching fields in Sanity (if you make an edit or addition in Sanity to a citation field that where that corresponding field is blank in Zotero for that citation, it will persist as long as it isn't overridden in Zotero, but we recommend treating Zotero as the source of truth for all citations).

This has pruned 52 unused citation records and 33 creators from the database, many of which contained out-of-date data with regards to how they represented their relationship to topics, which was caused the `Item type topic not defined for this list` warning you referenced above @gt150. Out of an abundance of caution, topics are not being automatically deleted but there were only about 6 candidates for pruning and I will be reviewing them manually.
